lib/gateway/Shard.js

```diff
@@ -2071,6 +2071,222 @@ class Shard extends EventEmitter {
+                    };
+                }
+                member = channel.members.update(packet.d, this.client);
+                /**
+                * Fired when a thread member is updated
+                * @event Client#threadMemberUpdate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The channel
+                * @prop {ThreadMember} member The updated thread member
+                * @prop {Object} oldMember The old thread member data
+                * @prop {Number} oldMember.flags User thread settings
+                */
+                this.emit("threadMemberUpdate", channel, member, oldMember);
+                break;
+            }
+            case "THREAD_MEMBERS_UPDATE": {
+                const channel = this.client.getChannel(packet.d.id);
+                if(!channel) {
+                    this.emit("debug", `Missing channel ${packet.d.id} in THREAD_MEMBERS_UPDATE`);
+                    break;
+                }
+                channel.update(packet.d);
+                let removedMembers;
+                let addedMembers;
+                if(packet.d.removed_member_ids) {
+                    removedMembers = packet.d.removed_member_ids.map((id) => channel.members.remove({id}) || {id});
+                }
+                if(packet.d.added_members) {
+                    addedMembers = packet.d.added_members.map((m) => {
+                        if(m.presence) {
+                            m.presence.id = m.presence.user.id;
+                            this.client.users.update(m.presence.user, this.client);
+                        }
+
+                        m.thread_id = m.id;
+                        m.id = m.user_id;
+                        m.member.id = m.member.user.id;
+                        const guild = this.client.guilds.get(packet.d.guild_id);
+                        if(guild) {
+                            guild.members.update(m.presence, guild);
+                            guild.members.update(m.member);
+                        }
+                        return channel.members.update(m, this.client);
+                    });
+                }
+                /**
+                * Fired when anyone is added or removed from a thread. If the `guildMembers` intent is not specified, this will only apply for the current user
+                * @event Client#threadMembersUpdate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The thread channel
+                * @prop {Array<ThreadMember | Object>} removedMembers An array of members that were removed from the thread channel. If a member is uncached, it will be an object with an `id` key. No other property is guaranteed
+                * @prop {Array<ThreadMember>} addedMembers An array of members that were added to the thread channel
+                */
+                this.emit("threadMembersUpdate", channel, removedMembers, addedMembers);
```

Comment:
Either this or the types need to be changed, types say both `addedMembers` and `removedMembers` are always present, and the current behavior makes them able to be undefined.

Reply:
Will reconsider how to structure this event